### ISSUE TITLE
Option to ignore file extension and detect archive format by file content

### ISSUE THREAD

**Issue by yene** [mholt/archiver#255]
## What would you like to have changed?
An option to ignore the file extension and detect the archive format based on the file content. 

## Why is this feature a useful, necessary, and/or important addition to this project?
Some of our customers deliver file formats with wrong extensions.

## What alternatives are there, or what are you doing in the meantime to work around the lack of this feature?
No idea. Maybe a bash file -I to check the type.

**Comment by coolaj86** [comment]
There's now a `--ext` option which you can use set set the type explicitly, as per https://github.com/mholt/archiver/pull/252, released with v3.4.0.

I don't think that we're going to do any type of sniffing on files, but you could do something like this, if it's a script environment:

```bash
arc -ext rar unarchive ~/Downloads/bat-v0.15.4-x86_64-apple-darwin.tar.gz
arc -ext tar unarchive ~/Downloads/bat-v0.15.4-x86_64-apple-darwin.tar.gz
arc -ext tar.bz2 unarchive ~/Downloads/bat-v0.15.4-x86_64-apple-darwin.tar.gz
arc -ext tar.gz unarchive ~/Downloads/bat-v0.15.4-x86_64-apple-darwin.tar.gz
arc -ext tar.xz unarchive ~/Downloads/bat-v0.15.4-x86_64-apple-darwin.tar.gz
arc -ext zip unarchive ~/Downloads/bat-v0.15.4-x86_64-apple-darwin.tar.gz
```

The best way to do that would be to try each

**Comment by coolaj86** [comment]
You could also use `file` to help determine the file type by "magic headers":

```bash
file /tmp/evil/double-evil.rar
/tmp/evil/double-evil.rar: POSIX tar archive
```

**Comment by coolaj86** [comment]
@yene Are the proposed workarounds sufficient to close the issue?

**Comment by mholt** [comment]
This has been implemented for all formats in #302, which will become v4 soon. Please give it a try and let me know your thoughts!
